app/src/main/java/com/vero/tasky/agenda/data/repository/AgendaRepositoryImpl.kt

```diff
@@ -40,49 +40,45 @@ class AgendaRepositoryImpl(
     }
 
     override suspend fun syncAgenda(
-        deletedEventsIds: List<String>,
-        deletedTasksIds: List<String>,
-        deletedReminderIds: List<String>
-    ): Result<Unit> = ApiCallHandler.safeApiCall {
+        deletedEventIds: Array<String>,
+        deletedTaskIds: Array<String>,
+        deletedReminderIds: Array<String>
+    ): Result<Unit> = safeApiCall {
         api.syncAgenda(
             SyncAgendaRequest(
-                deletedEventsIds = deletedEventsIds,
-                deletedTasksIds = deletedTasksIds,
+                deletedEventIds = deletedEventIds,
+                deletedTaskIds = deletedTaskIds,
                 deletedReminderIds = deletedReminderIds
             )
         )
     }
 
-    override suspend fun getFullAgenda() = ApiCallHandler.safeApiCall {
+    override suspend fun getFullAgenda() = safeApiCall {
         val result = api.getFullAgenda()
         saveAgendaItems(result)
     }
 
     private fun getAgendaItemsFromDao(timestamp: Long) : List<AgendaItem> {
         val agendaItems = mutableListOf<AgendaItem>()
-        agendaItems.addAll(taskDao.loadTasksForDay(timestamp).map {
-            TaskMapper.toTask(it)
-        })
-        agendaItems.addAll(eventDao.loadEventsForDay(timestamp).map {
-            EventMapper.toEvent(it)
-        })
-        agendaItems.addAll(reminderDao.loadRemindersForDay(timestamp).map {
-            ReminderMapper.toReminder(it)
-        })
+        agendaItems.addAll(taskDao.loadTasksForDay(timestamp).map { it.toTask() })
+        agendaItems.addAll(eventDao.loadEventsForDay(timestamp).map {it.toEvent() })
+        agendaItems.addAll(reminderDao.loadRemindersForDay(timestamp).map { it.toReminder() })
         return agendaItems
     }
 
     private suspend fun saveAgendaItems(agendaDto: AgendaDto) {
         supervisorScope {
+            val jobsList = mutableListOf<Job>()
```

Comment:
Same as what I mentioned in my previous comments

You can have a mutableList, or just save the jobs in variables (val events = agenda... .launch), and then just return the join
return (events + tasks + reminders).joinAll()